### PR TITLE
Factorial refactor changes

### PR DESCRIPTION
**Pull request by vyark**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -25,5 +25,42 @@
...
+        </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-autoconfigure</artifactId>
+            <version>2.6.3</version>
+        </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-web</artifactId>
+            <version>2.6.3</version>
```
Comment (fedorchuck) [inline pom.xml]: Overall is good idea to move version separate like
<version>${groupId.artefactId.version}</version>

File: src/main/java/filescanner/FileScanner.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+import java.io.File;
+import java.util.concurrent.RecursiveAction;
+
+public class FileScanner extends RecursiveAction {
+
+    @Override
+    protected void compute() {
+        DirectoryScanner directoryScanner = new DirectoryScanner();
+        directoryScanner.setBasedir(new File("C:\\Users\\Volha_Yarkouskaya"));
```
Comment (fedorchuck) [inline src/main/java/filescanner/FileScanner.java]: good practice to move such paths into params or constructor

File: src/main/java/filescanner/ProgressBarAction.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    protected void compute() {
+        char[] animationChars = new char[]{'|', '/', '-', '\\'};
+
+        for (int i = 0; i <= 100; i++) {
+            System.out.print("Processing: " + i + "% " + animationChars[i % 4] + "\r");
+
+            try {
+                Thread.sleep(200);
+            } catch (InterruptedException e) {
+                e.printStackTrace();
```
Comment (fedorchuck) [inline src/main/java/filescanner/ProgressBarAction.java]: such practice is really bad. Please thrown exception, or log it, or wrap to non-checked or use lombok @Sneakythrows(InterruptedException .class) .
Please let me know if you interested on the reason

File: src/main/java/filescanner/ProgressBarAction.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+import java.util.concurrent.RecursiveAction;
+
+public class ProgressBarAction extends RecursiveAction {
+    @Override
+    protected void compute() {
+        char[] animationChars = new char[]{'|', '/', '-', '\\'};
+
+        for (int i = 0; i <= 100; i++) {
+            System.out.print("Processing: " + i + "% " + animationChars[i % 4] + "\r");
```
Comment (fedorchuck) [inline src/main/java/filescanner/ProgressBarAction.java]: you may use lombok @Slf4j to print any output to the console

File: src/main/java/filescanner/SizeAction.java [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+    public SizeAction(DirectoryScanner directoryScanner) {
+        this.directoryScanner = directoryScanner;
+    }
+
+    @Override
+    protected void compute() {
+        int filesSize = 0;
+        for (String fileName : directoryScanner.getIncludedFiles()) {
+            filesSize += directoryScanner.getResource(fileName).getSize();
```
Comment (fedorchuck) [inline src/main/java/filescanner/SizeAction.java:22]: are you able to write a unit test to check that it's work fine with current implementation?)

File: src/main/java/rest/CompletableFutureService.java [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpUriRequest;
+import org.apache.http.impl.client.HttpClientBuilder;
+
+import java.io.IOException;
+import java.net.http.HttpResponse;
+
+public class CompletableFutureService {
+    public static void main(String[] args) throws IOException {
```
Comment (fedorchuck) [inline src/main/java/rest/CompletableFutureService.java]: do we need this class? can we replace it for unit test?

File: src/main/java/rest/EmployeeServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+        employee.setId(1L);
+        employee.setName("Olga");
+        employee.setSalary(new BigInteger("5000"));
+
+        return Arrays.asList(employee);
+    }
+
+    @Override
+    public BigInteger getSalary(Long id) {
+        return null;
```
Comment (fedorchuck) [inline src/main/java/rest/EmployeeServiceImpl.java]: I'm a bit confused: hired employees has an employee but salary is null. is it okay?

File: src/main/java/rest/EmployeeService.java [hunk 8]
```diff
@@ -0,0 +1,10 @@
+package rest;
+
+import java.io.IOException;
+import java.math.BigInteger;
+import java.util.List;
+
+public interface EmployeeService {
+    List<Employee> hiredEmployees() throws IOException;
+    BigInteger getSalary(Long id);
```
Comment (fedorchuck) [inline src/main/java/rest/EmployeeService.java]: I'm not sure about naming - whom salary this method return? can salary be like 123.123 ?

File: src/test/java/filescanner/FileScannerTest.java [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+import org.junit.Test;
+
+import java.util.concurrent.ForkJoinPool;
+
+public class FileScannerTest {
+
+    private FileScanner fileScanner = new FileScanner();
+
+    @Test
+    public void test() {
```
Comment (fedorchuck) [inline src/test/java/filescanner/FileScannerTest.java:12]: what this test test?)

**Review by fedorchuck — COMMENTED** [review]
I'd suggest to implement comments

**Comment by vyark** [comment]
Sorry, rest test is not completed. Only factorial, file scanner and quicksort are completed tasks.

File: src/main/java/filescanner/SizeAction.java (hunk 5 above)
Comment (vyark) [reply src/main/java/filescanner/SizeAction.java:22]: Done.

File: src/main/java/rest/CompletableFutureService.java (hunk 6 above)
Comment (vyark) [reply src/main/java/rest/CompletableFutureService.java]: Class was removed.

File: src/main/java/rest/EmployeeServiceImpl.java (hunk 7 above)
Comment (vyark) [reply src/main/java/rest/EmployeeServiceImpl.java]: Class was removed.

**Pull request merged**